### PR TITLE
[Feature] Issue 245 - Variable Image Width

### PR DESCRIPTION
**Pull request by loayza**
This is an improved reattempt of [this other pull request](https://github.com/woothemes/FlexSlider/pull/877).

I realised that the mentioned request got stucked due a [merge conflict](https://github.com/woothemes/FlexSlider/pull/877#js-pull-merging) that was never resolved and it was apparently abandoned by the author.

What I've done is to fork the current version, to resolve the conflict and to rebase onto the latest version. 

I've tried to contact the author of the original request before creating this new request, but I've received no reply from him. 

Nevertheless, I suppose that he would be happy if his code is finally added to the plug-in :-)

### CODE REVIEW

**Comment by wells** [comment]
@loayza I hope this does get commited.

**Comment by zanematthew** [comment]
@loayza @wells I noticed this one also has issues, in that it doesn't advance the slider forward when used in thumbs. Here's a working example of the default slider with thumbs http://flexslider.woothemes.com/thumbnail-slider.html

Granted this will untouched/noticed for several more months.

File: jquery.flexslider.js [hunk 1]
```diff
@@ -568,15 +571,34 @@
...
-            slider.newSlides.width(slider.computedW);
-            slider.setProps(slider.computedW, "setTotal");
+            slider.setProps(slider.computedW, "setCurrent");
           }
         }
       },
+      variableImageWidth: function() {
+        var minHeight = 0;
+        slider.newSlides.each(function(index) {
+          $(this).children('img').first().height('auto');
```
Comment (jsit) [inline jquery.flexslider.js]: Why only use `img` elements to determine slide height? What if a slide has no `img` element?

**Comment by jsit** [comment]
Should this setting be named something else? Not all slides will necessarily be images; some may be just text, or an image with a caption, or a YouTube embed, etc.

File: jquery.flexslider.js [hunk 2]
```diff
@@ -833,9 +856,43 @@
...
+                } else if (slider.currentSlide === slider.last && slider.animatingTo === 0 && slider.vars.animationLoop && slider.direction !== "prev") {
+                  offset = (reverse) ? 0 : slider.count + 1;
+                }
+
+                slider.newSlides.each(function(index) {
+                  var imgWidth = $(this).children('img').first().width();
+                  if(imgWidth === 0 || imgWidth === null) imgWidth = slider.computedW;
+                  if(index == current) slider.currentWidth = imgWidth;
+                  if(index < start) startWidth += imgWidth;
+                  if(index < offset) offsetWidth += imgWidth;
```
Comment (jentscke) [inline jquery.flexslider.js]: Here you should additionally sum slider.vars.itemMargin to get itemMargin working correctly

**Pull request closed without merging**